test/run_test.py

```diff
@@ -898,19 +900,21 @@ def print_log_file(test: str, file_path: str, failed: bool) -> None:
 
 
 def get_pytest_args(options, stepcurrent_key, is_cpp_test=False):
-    if RERUN_DISABLED_TESTS:
-        # When under rerun-disabled-tests mode, run the same tests multiple times to determine their
-        # flakiness status. Default to 50 re-runs
-        rerun_options = ["--flake-finder", "--flake-runs=50"]
-    elif options.continue_through_error:
-        # If continue through error, don't stop on first failure
-        rerun_options = ["--reruns=2"]
-    else:
-        # When under the normal mode, retry a failed test 2 more times. -x means stop at the first
-        # failure
-        rerun_options = ["-x", "--reruns=2"]
-        if IS_CI:
+    if IS_CI:
+        if RERUN_DISABLED_TESTS:
+            # When under rerun-disabled-tests mode, run the same tests multiple times to determine their
+            # flakiness status. Default to 50 re-runs
+            rerun_options = ["--flake-finder", "--flake-runs=50"]
+        elif options.continue_through_error:
+            # If continue through error, don't stop on first failure
+            rerun_options = ["--reruns=2"]
+        else:
+            # When under the normal mode, retry a failed test 2 more times. -x means stop at the first
+            # failure
+            rerun_options = ["-x", "--reruns=2"]
```

Comment:
Nit: We could have this as an array now, i.e. `["-x", "--reruns=2",  f"--sc={stepcurrent_key}"]`